core/services/workflows/engine.go

```diff
@@ -583,6 +637,13 @@ type Config struct {
 	MaxWorkerLimit   int
 	QueueSize        int
 	NewWorkerTimeout time.Duration
+	DONInfo          *DONInfo
+}
+
+type DONInfo struct {
```

Comment:
Any change we could use the one from remote/types/typed.go? That one will become part of the Registry soon. You could add shared secret there and pass PeerID separately somehow.